sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ContactDirectorySteps.java

```diff
@@ -209,9 +213,17 @@ public ContactDirectorySteps(
 
     When(
         "I apply Id of last api created Contact on Contact Directory Page",
+        () -> {
+          String contactUuid = apiState.getCreatedContact().getUuid();
+          webDriverHelpers.fillAndSubmitInWebElement(
+              CONTACT_DIRECTORY_DETAILED_PAGE_FILTER_INPUT, contactUuid);
+        });
+
+    When(
+        "I apply Id of last created Contact on Contact Directory Page",
         () -> {
           String contactUuid =
-                  apiState.getCreatedContact().getUuid();
+              dataOperations.getPartialUuidFromAssociatedLink(collectedContact.getUuid());
```

Comment:
better would be to fulfill filter field with whole UUID not partial, can cause problems with mutliple records in a grid